prereise/gather/demanddata/transportation_electrification/immediate.py

```diff
@@ -0,0 +1,194 @@
+
+    TRANS_charge = np.zeros(24 * len(input_day))
+
+    data_day = mileage.get_data_day(newdata)
+
+    actual_vmt = 0
+    potential_vmt = 0
+
+    # from grid to battery efficiency
+    charging_efficiency = 0.9
+
+    # charging when SOC less than batterstrategy
+    battery_strategy = 1
+
+    # 1-anytrip number, 2-last trip
+    trip_strategy = 1
+
+    kwh = kwhmi * veh_range
+
+    for day_iter in range(len(input_day)):
+
+        electricload = np.zeros(4800)
+        # initializes electricload_old
+
+        # vmt, trip amount, vehicle amount
+        info1 = [0] * 3
+
+        # meh: better names for counters
+        trip_num = 1  # trip number for current vehicle
+        # charging start and end points
+        start_point = 0
+        end_point = 0
+
+        # flag to see if the trip is the first for the vheicle, '1' yes, '0' no
+        firstrip = 0
+
+        # meh: day2 = data_day, month2 = data_month
+        for i in range(len(newdata)):
+            if data_day[i] == input_day[day_iter]:
+                if (
+                    i > 0
+                    and newdata.loc[i]["sample vehicle number"]
+                    == newdata.loc[i - 1]["sample vehicle number"]
+                ):
+                    newdata.loc[i]["trip start battery charge"] = (
+                        newdata.loc[i - 1]["trip end battery charge"]
+                        + newdata.loc[i - 1]["depleting time"]
+                    )
+                    # trip number
+                    trip_num += 1
+                    newdata.loc[i]["trip number"] = trip_num
+                else:
+                    newdata.loc[i]["trip start battery charge"] = kwh
+                    trip_num = 1
+                    # trip number
+                    newdata.loc[i]["trip number"] = trip_num
+
+                # battery electric vehicle
+                # pure electric
+                ER = 1
+
+                # 1 is the safety coefficient
+                if newdata.loc[i]["total vehicle miles traveled"] < veh_range * 1:
+                    # 1 means the day trip could be used in battery electric vehicle
+                    newdata.loc[i]["BEV could be used"] = 1
+                    # trip end battery charge
+                    newdata.loc[i]["trip end battery charge"] = (
+                        newdata.loc[i]["trip start battery charge"]
+                        - newdata[i]["Miles traveled"] * kwhmi * ER
+                    )
+                    # period when battery is discharging. depleting time
+                    newdata.loc[i]["depleting time"] = newdata.loc[i][
+                        "Travel time (hour decimal)"
+                    ]
+
+                else:
+                    # 0 means the day trip could not be used in battery electric vehicle
+                    newdata.loc[i]["BEV could be used"] = 0
+
+                # fuel consumption is 0
+                newdata.loc[i]["fuel consumption"] = 0

```

Comment:
If this is the same for every entry, can it be assigned to the whole column outside of the loop?

Reply:
This should actually be deleted.